### PR TITLE
Fix C comments and dependency on spaces.

### PR DESCRIPTION
**Pull request by Spacetown**
As discussed in #525 C comments aren't ignored in `_is_non_code` and there is a dependency on spaces in `} else {` (see extended tests). A empty line or a line with a C++ coment is treated as non code but a C comment is treated as code and decreases the coverage. 

[no changelog]

### CODE REVIEW

**Comment by Spacetown** [comment]
Failed test is OK. The lines `} // GCOV_EXCL_STOP` and `} // LCOV_EXCL_STOP´ are now `non_code`. In the previous version only a emty `}` was treated as `non_code`.

File: gcovr/gcov_parser.py [hunk 1]
```diff
@@ -1029,20 +1029,30 @@ def _is_non_code(code: str) -> bool:
...
     False
     """
 
-    # This variant is more flexible, but would change JSON reports
-    # code = code.replace("{", "").replace("}", "").strip()
-    code = code.strip().replace("{", "").replace("}", "")
-    return len(code) == 0 or code.startswith("//") or code == "else"
+    code = _CPP_STYLE_COMMENT_PATTERN.sub("", code)
+    code = _C_STYLE_COMMENT_PATTERN.sub("", code)
+    code = code.replace(" ", "").strip()
```
Comment (latk) [inline gcovr/gcov_parser.py:1054]: The replace() is useless in this context.

```suggestion
    code = code.strip()
```

**Review by latk — COMMENTED** [review]
Extremely good change, especially since the excl-branch test was previously wrong (claiming that a line was excluded although it wasn't).

However, the changes to the doc/examples seem out of place in this context. If the files should be removed, that should be discussed as part of #543.

**Comment by Spacetown** [comment]
> However, the changes to the doc/examples seem out of place in this context. If the files should be removed, that should be discussed as part of #543.

That was an commit by accident, I will remove the changes.

**Review by latk — APPROVED** [review]

**Pull request merged**